lib/spack/spack/test/build_environment.py

```diff
@@ -465,7 +465,7 @@ def test_setting_dtags_based_on_config(config_setting, expected_flag, config, mo
     pkg = s.package
 
     env = EnvironmentModifications()
-    with spack.config.override("config:shared_linking", config_setting):
+    with spack.config.override("config:shared_linking", {"type": config_setting, "bind": False}):
```

Comment:
test should handle both cases (string and type/bind dict)

Reply:
This is updating old syntax internally, there's a test elsewhere that ensures old syntax `override(config:shared_linking:runpath)` works too.